lib/job.jsonnet

```diff
@@ -301,7 +301,7 @@ function(jobName, agentEnv={}, stepEnvFile='', patchFunc=identity) patchFunc({
     annotations: annotations,
   },
   spec: {
-    backoffLimit: 0,
+    backoffLimit: 2,
```

Comment:
Let's make this an opt-in behavior and configurable like with the `BUILDKITE_PLUGIN_K8S_JOB_TTL_SECONDS_AFTER_FINISHED` config - so we don't by default retry a command that fails midway when its not necessarily safe to do so